primeqa/components/reader/generative.py

```diff
@@ -204,7 +204,7 @@ def predict(
             )
         )
 
-        self._preprocessor.max_contexts = len(processed_context)
+        self._preprocessor._max_contexts = len(processed_context)
```

Comment:
@mihaela-bornea I recommend adding a setter function to set the value of `private` member like `_max_contexts`. 

PEP8 naming convention implies that any variable with leading underscore should be considered private and preferably not directly accessed outside.

Reply:
Done !